src/msspreadsheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,639 @@
+        json payload = {"name": worksheetName};
+        request.setJsonPayload(payload);
+        http:Response|error response = self.workbookClient->post("/v1.0/me/drive/root:" +
+                    self.properties.path + self.properties.workbookName + ".xlsx:/workbook/worksheets", request);
+
+        if response is error {
+            HttpError httpError = error(HTTP_ERROR, message = "Error occurred while accessing the Microsoft Graph API.", 
+                                        errorCode = HTTP_ERROR, cause = response);
+            return httpError;
+        }
+
+        http:Response httpResponse = <http:Response> response;
+
+        if httpResponse.statusCode != http:STATUS_CREATED {
+            HttpResponseHandlingError httpResponseHandlingError = error(HTTP_RESPONSE_HANDLING_ERROR,
+                message = "Error occurred while creating the worksheet.", errorCode = HTTP_RESPONSE_HANDLING_ERROR);
+            return httpResponseHandlingError;
+        }
+
+        //If the worksheet was created we will get a JSON response with the newly created worksheet's information
+        json|error responseJson = httpResponse.getJsonPayload();
+
+        if !(responseJson is map<json>) {
+            typedesc<any|error> typeOfRespone = typeof responseJson;
+            TypeConversionError typeError = error(TYPE_CONVERSION_ERROR, 
+                    message = "Invalid response; expected a `map<json>` found " +  typeOfRespone.toString(), 
+                    errorCode = TYPE_CONVERSION_ERROR);
+            return typeError;
+        }
+
+        map<json> responsePayload = <map<json>> responseJson;
+    
+        json|error identifier = responsePayload.id;
+
+        if !(identifier is string) {
+            typedesc<any|error> typeOfIdentifier = typeof identifier;
+            TypeConversionError typeError = error(TYPE_CONVERSION_ERROR, 
+                    message = "Invalid identifier; expected a `string` found " +  typeOfIdentifier.toString(), 
+                    errorCode = TYPE_CONVERSION_ERROR);
+            return typeError;
+        }
+
+        string sheetId = <string> identifier;
+
+        json|error sheetPosition = responsePayload.position;
+
+        if !(sheetPosition is int) {
+            typedesc<any|error> typeOfPosition = typeof sheetPosition;
+            TypeConversionError typeError = error(TYPE_CONVERSION_ERROR, 
+                    message = "Invalid sheet position; expected an `int` found " +  typeOfPosition.toString(), 
+                    errorCode = TYPE_CONVERSION_ERROR);
+            return typeError;
+        }
+
+        int position = <int> sheetPosition;
+        
+        //Populate a new worksheet object using the information from the properties as well as from the received JSON object
+        Worksheet workSheet = new(self.microsoftGraphConfig, self.properties.path,
+                                self.properties.workbookName, sheetId, worksheetName, position);
```

Comment:
Wrong indentation?

Reply:
Did reformatting the code in all the bal files